src/main/java/com/google/sps/app/servlets/SignUp.java

```diff
@@ -0,0 +1,52 @@
+package com.google.sps.app.servlets;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.gson.Gson;
+import com.google.sps.util.UUIDs;
+import com.google.sps.util.Validator;
+
+@WebServlet("/SignUp")
+public class SignUp extends HttpServlet{
+
+    final String textInput = "text-input-";
```

Comment:
You can also move the table read write to a separate class. you can do that in a follow up PR if you prefer

class UserDBHelper {
  private static final String USER_NAME_PARAM  =
    "text-input-user-name";
  readFromBD()
  writeToDB()
}